boefjes/boefjes/katalogus/plugins.py

```diff
@@ -70,6 +70,12 @@ def list_plugins(
     # filter plugins by scan level for boefje plugins
     plugins = [plugin for plugin in plugins if plugin.type != "boefje" or plugin.scan_level >= filter_params.scan_level]
 
+    if filter_params.consumes is not None:
+        plugins = [plugin for plugin in plugins if filter_params.consumes.issubset(set(plugin.consumes))]
+
+    if filter_params.produces is not None:
+        plugins = [plugin for plugin in plugins if filter_params.produces.issubset(set(plugin.produces))]
```

Comment:
Does this actually allow for plugins that only produce one of the filtered produce values?
Eg, I filter on produces, X or Y. this plugin produces Y and Z.
X and y is not a subset of Y and Z. It does produce an intersect result which Is probably what we are looking for as filters are usually OR, not AND?

Reply:
```suggestion
        plugins = [plugin for plugin in plugins if filter_params.produces.intersection(set(plugin.produces))]
```

Reply:
Perhaps not relevant to the discussion, but the scheduler will only request one type.

Reply:
The idea when writing this was to do an AND, but an OR indeed makes more sense

Reply:
Doing an AND would change the behaviour of boefjes I'd guess?

Reply:
How so? Given that the scheduler only uses one element in its queries, both implementations would have the same result from the schedulers point of view